src/consumer.ts

```diff
@@ -165,14 +171,57 @@ export class Consumer extends EventEmitter {
     this.stopped = true;
   }
 
+  public async reportMessageFromBatchFinished(message: SQSMessage, error: Error): Promise<void> {
+    debug('Message from batch has finised');
+
+    this.concurencyLimit++;
+    this.updateBatchSize();
+
+    try {
+      if (error) throw error;
+
+      await this.deleteMessage(message);
+      this.emit('message_processed', message);
+    } catch (err) {
+      this.emitError(err, message);
+    }
+  }
+
+  private reportNumberOfMessagesReceived(numberOfMessages: number): void {
+    debug('decrementing next batch size');
+    this.concurencyLimit = this.concurencyLimit - numberOfMessages;
+    this.updateBatchSize();
+  }
+
+  private updateBatchSize(): void {
+    debug('Updating next batch size');
+
+    this.batchSize = Math.min(10, this.concurencyLimit);
+
+    // instrument current batch size
+    if (this.batchSizeUpdatedInstrument) {
+      this.batchSizeUpdatedInstrument({
+        instanceId: process.env.HOSTNAME,
+        queueUrl: this.queueUrl,
+        currentConcurencyLimit: this.concurencyLimit,
```

Comment:
now i understand the mixup, you are missing a 'nextBatchSize' variable, which is the one that should be changed throughout the polling process. the concurrencyLimit variable should be constant and passed during initialization, just like batchSize is passed to the old infra poller